services/storage/series_cursor.go

```diff
@@ -181,12 +183,16 @@ RETRY:
 		c.row.stags = sr.Tags
 		c.tags = copyTags(c.tags, sr.Tags)
 
-		c.filterset = mapValuer{"_name": string(sr.Name)}
+		c.filterset = make(mapValuer)
 		for _, tag := range c.tags {
 			c.filterset[string(tag.Key)] = string(tag.Value)
 		}
 
-		c.tags.Set(measurementKey, sr.Name)
+		if !c.multiTenant {
+			c.filterset["_name"] = string(sr.Name)
+			c.tags.Set(measurementKey, sr.Name)
```

Comment:
If it is multiTenant is the mesurementKey already part of the tags?

Reply:
exactly – which means less allocations and more efficient 🎉

Reply:
…and now we just need to get `_field` in there

Reply:
Makes sense